### PR TITLE
npm/yarn audit: fix CWE parsing

### PR DESCRIPTION
**Pull request by valentijnscholten**
`npm audit` reports are suffering from the same thing as `yarn audit` reports in #5999

CWE values are references for an advisory:
```
          "cwe":"[\"CWE-918\",\"CWE-1333\"]",
```
After fixing that, the format changed without warning into:
```
        "cwe": [
          "CWE-346",
          "CWE-453"
        ],
```

So now this PR supports all 3 possible formats for both `yarn audit` and `npm audit`.
New example files are real life examples.

### CODE REVIEW

**Comment by valentijnscholten** [comment]
hmmm after fixing this and making the unit tests pass, it seems the output has changed again. now it is a proper list:

```
        "cwe": [
          "CWE-346",
          "CWE-453"
        ],
```

let's wait a couple of days to see what else they have up their sleeves :-)

**Comment by damiencarol** [comment]
> let's wait a couple of days to see what else they have up their sleeves :-)

XD

File: dojo/tools/utils.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            cwe_string = cwe_node[4:]
+            if cwe_string:
+                return int(cwe_string)
+        elif cwe_node.startswith('['):
+            cwe = json.loads(cwe_node)
+            if cwe:
+                return int(cwe[0][4:])
+
+    # Use CWE-1035 as fallback (vulnerable third party component)
+    return 1035
```
Comment (damiencarol) [inline dojo/tools/utils.py:30]: This choice is not always true. If we want to use this function for all parsers, sometimes it's not component-oriented. could we just return an empty array?

Comment (valentijnscholten) [reply dojo/tools/utils.py:30]: this function is for npm/yarn only

**Comment by StefanFl** [comment]
@valentijnscholten Which version of `npm` do you use? With the most recent version 8.5.4 I get a different looking JSON file without the advisories.

**Comment by valentijnscholten** [comment]
Npm 6. Anything above that is not supported, https://github.com/DefectDojo/django-DefectDojo/issues/4285

**Review by StefanFl — APPROVED** [review]

**Review by damiencarol — APPROVED** [review]

**Pull request merged**